### PR TITLE
glib-utils: remove n_fields function

### PR DESCRIPTION
**Pull request by shlyakpavel**
Ported https://github.com/mate-desktop/engrampa/pull/358
Closes https://github.com/lxqt/lxqt-archiver/issues/180

### CODE REVIEW

**Comment by shlyakpavel** [comment]
I have no idea how to test that, so check twice haha
The code was written due to some vulnerabilities (or problems?) that I found in 2020, I don't remember any details

File: src/core/glib-utils.c [hunk 1]
```diff
@@ -418,7 +418,7 @@ _g_strdup_with_max_size (const char *s,
 		char *first_half;
 		char *second_half;
 		int   offset;
-		int   half_max_size = max_size / 2 + 1;
+        int   half_max_size = max_size / 2 + 1;n_fields
```
Comment (tsujan) [inline src/core/glib-utils.c:421]: Although this typo is commented out, please reverse it.

**Review by tsujan — CHANGES_REQUESTED** [review]

**Comment by tsujan** [comment]
> I have no idea how to test that,

There's no need to a test. `g_strv_length()` exactly does what `n_fields()` did. The difference is only that `n_fields()` didn't need a nullity check (an "advantage").

File: src/core/glib-utils.c (hunk 1 above)
Comment (shlyakpavel) [reply src/core/glib-utils.c]: Done

**Comment by shlyakpavel** [comment]
I'll squash commits, please wait

**Review by tsujan — APPROVED** [review]
Thanks!

**Pull request merged**